### PR TITLE
interfaces/builtin: reduce duplication and remove cruft in Sanitize{Plug,Slot}

### PR DESCRIPTION
**Pull request by zyga**
This branch unifies how Sanitize{Plug,Slot} are implemented across all of the interfaces.
This removes a lot of copy-pasted code and adds some missing calls in other copy-pasted code.
Lastly, because all of the error messages are in a small set of four functions, there is much more
uniformity in how error messages look like.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: interfaces/builtin/utils.go [hunk 1]
```diff
@@ -91,3 +91,39 @@ func udevUsbDeviceSnippet(subsystem string, usbVendor int64, usbProduct int64, k
 func udevSnapSecurityName(snapName string, appName string) string {
 	return fmt.Sprintf(`snap_%s_%s`, snapName, appName)
 }
+
+func ensureSlotIfaceMatch(iface interfaces.Interface, slot *interfaces.Slot) {
```
Comment (mvo5) [inline interfaces/builtin/utils.go]: Do we need (explicit) tests for the new helpers?

Comment (zyga) [inline interfaces/builtin/utils.go]: They are tested indirectly but I'll add a small set of tests. Thanks for spotting that.

Comment (zyga) [inline interfaces/builtin/utils.go]: I just pushed some nice tests

**Comment by zyga** [comment]
After discussing this with @pedronis I'm going to move the responsibility of checking interface name from `Sanitize{Plug,Slot}` to the caller.

File: interfaces/core.go [hunk 2]
```diff
@@ -38,6 +38,22 @@ func (plug *Plug) Ref() PlugRef {
...
 }
 
+// Sanitize plug with a given snapd interface.
+func (plug *Plug) Sanitize(iface Interface) error {
+	if iface.Name() != plug.Interface {
+		return fmt.Errorf("cannot sanitize plug %q (interface %q) using interface %q",
+			plug.Ref(), plug.Interface, iface.Name())
+	}
+	type sanitizer interface {
+		SanitizePlug(plug *Plug) error
```
Comment (chipaca) [inline interfaces/core.go]: I'd make these interfaces public, to better document the thing (much as for example `http.Flusher` is public to document that). It adds a path to discovery of this feature.

Comment (zyga) [inline interfaces/core.go]: Done.

**Pull request merged**